articles/WideStrings.md

```diff
@@ -0,0 +1,264 @@
+one character with the next in non-trivial ways.  As one of the goals of ROS 2.0
+is to support small, constrained systems, dealing with wide strings may be out of
+the question.  Thus, the current ROS 2.0 design for wide strings makes them optional,
+defines them as a separate type from regular strings, and recommends against
+using them for common messages.
+
+### What is the encoding of the wide string?
+
+There are two main ways that ROS 2.0 can provide for the encoding of a wide string.  Either
+ROS 2.0 can define exactly what the encoding will be for all strings, or it can allow the
+user to specify the encoding when handing it to ROS 2.0.  There are pros and cons to each
+approach.
+
+**ROS 2.0 defines the encoding**
+
+Pros:
+
+* All strings known to be in the same encoding.
+
+Cons:
+
+* If the user-level code uses a different encoding, the user is responsible
+for doing a conversion to the ROS 2.0 defined encoding.  This can involve
+some runtime cost.
+
+**User defines the encoding**
+
+Pros:
+
+* No conversions necessary for user-to-ROS 2 API.
+
+Cons:
+
+* The user must always tell ROS 2.0 what encoding the data is in.
+* The encoding needs to be transmitted on the wire to the other side.
+* It is not clear how ROS 2.0 will transmit the encoding.  Fixed list?  Which
+encodings go into the list?
+
+While a user defined encoding looks somewhat attractive at first, the downsides
+to it make it difficult to determine exactly which encodings are in use.  The
+downsides also mean that any program that wants to parse ROS 2 messages (or bag
+files) may potentially have to deal with many different encodings.  Additionally,
+Unicode (and in particular, UTF-8) are very commonplace now, while other encodings
+are becoming more esoteric.  Even earlier Unicode (UTF-16 and UTF-32) encodings are
+essentially deprecated in favor of UTF-8.  For these reasons, the current design of
+the ROS 2.0 wide string is to use a UTF-8 encoding for all strings.
+
+### What does the API look like to a user of ROS 2.0?
+
+#### Python 3
+
+Python 3 splits byte arrays from strings pretty cleanly.  Thus, in Python 3, a
+string is a sequence of characters, where each character can take 1 or more bytes.
+Byte arrays are sequences of bytes.  Thus, the ROS 2.0 API for dealing with wide strings
+(such as the publish API) will take just a string type in, and encode it as utf-8.
+The ROS 2.0 API for dealing with regular strings will take a string type in, and
+encode it as ASCII.
+
+**Example**
+
+```
+import sys
+from time import sleep
+
+import rclpy
+
+from rclpy.qos import qos_profile_default
+
+from std_msgs.msg import WString
+
+def main(args=None):
+    if args is None:
+        args = sys.argv
+
+    rclpy.init(args)
+
+    node = rclpy.create_node('talker')
+
+    chatter_pub = node.create_publisher(WString, 'chatter', qos_profile_default)
+
+    msg = String()
+
+    i = 1
+    while True:
+        msg.data = 'Hello World: {0}'.format(i)
```

Comment:
If I understand correctly, this article proposes new ROS `std_msgs/WString`. If this is correct. I have impression that this is not convenient for those who uses non-English speaker. We'll consider both ASCII string and Non-ASCII string as "string"

For example, we'll write 
https://github.com/jsk-ros-pkg/trans_system/blob/88be85431823a9a7cdd5e846dcb887bb16fa844d/jaxon_ros_bridge/scripts/robot_command_server.py#L33-L35
```
        self.sound_pub.publish(SoundRequest(sound=SoundRequest.SAY,
                                            command=SoundRequest.PLAY_ONCE,
                                            arg="Servo off"))
```
And if we want robot to speak in Japanese, we have to change `SoundRequest` to different message type, that's feels strange to me. We might want to write something like
```
        if language_type == "English":
           text = "Servo off"
        else:
           text = "サーボオフ"
        self.sound_pub.publish(SoundRequest(sound=SoundRequest.SAY,
                                            command=SoundRequest.PLAY_ONCE,
                                            arg=text))
```

I think you have already discussed this issue and there might be huge drawback on this strategy...